avo-hq/avo
# Hide grid view cover if cover_url not defined (#2347)

# Description
At the moment the grid view cover handles two scenarios
- If cover_url is present, that url is used to display the cover image
- If not, a fallback image is used
We want to extend the second case distinguishing a not defined cover_url from a falsy value.

I noticed that the grid view cover empty state component was really similar to the code at link_to_cover.
So I took the initiative to merge them (the link_to_cover return did not have a wrapping div but I think this is fine 
because adding it does not affect anything visually).

Updated the avo docs to document how add custom styling to the cover. (https://github.com/avo-hq/avodocs/pull/165)

Fixes [# 2347](https://github.com/avo-hq/avo/issues/2347)

# Checklist:

## Screenshots & recording
![image](https://github.com/avo-hq/avo/assets/3640909/a7fac9c3-b846-4c1b-b020-f66dbae2ea46)
![image](https://github.com/avo-hq/avo/assets/3640909/4fa621d1-0873-4af8-9b74-92db19a04f0d)

## Manual review steps
1. Open Posts in grid view
2. Check if posts with image, show the image
3. Check if posts without image, show the fallback image (avocado)
4. Open Fish in grid view
5. Check if all posts have no cover

Manual reviewer: please leave a comment with output from the test if that's the case.
